### ISSUE TITLE
Content-collections is executed twice if next.config.ts is used

### ISSUE THREAD

**Issue by wottpal** [sdorra/content-collections#397]
Reopening #316. Tested with latest Next.js Canary and content-collection dependencies.

**Comment by sdorra** [comment]
I can't reproduce it. Which version of `@content-collections/next` do you use?

```bash
pnpm run dev

> next-canary@0.1.0 dev /Users/sdorra/Desktop/next-canary
> next dev

Starting content-collections content-collections.ts
build started ...
... finished build of 1 collection and 5 documents in 15ms
start watching ...
   ▲ Next.js 15.0.4-canary.8
   - Local:        http://localhost:3000
   - Network:      http://192.168.1.151:3000

 ✓ Starting...
 ✓ Ready in 757ms
 ```
 
 ```bash
 pnpm run build

> next-canary@0.1.0 build /Users/sdorra/Desktop/next-canary
> next build

Starting content-collections content-collections.ts
build started ...
... finished build of 1 collection and 5 documents in 15ms
   ▲ Next.js 15.0.4-canary.8

   Creating an optimized production build ...
 ✓ Compiled successfully
 ✓ Linting and checking validity of types
 ✓ Collecting page data
 ✓ Generating static pages (5/5)
 ✓ Collecting build traces
 ✓ Finalizing page optimization

Route (app)                              Size     First Load JS
┌ ○ /                                    136 B           100 kB
└ ○ /_not-found                          900 B           101 kB
+ First Load JS shared by all            100 kB
  ├ chunks/38765802-a0393788968c3986.js  52.5 kB
  ├ chunks/897-dd0b8668a59460c1.js       45.8 kB
  └ other shared chunks (total)          1.87 kB


○  (Static)  prerendered as static content
```

**Comment by wottpal** [comment]
> Which version of @content-collections/next do you use?

`0.2.3`

Can you try using `createContentCollectionPlugin` in your `next.config.ts` file with a different configuration file path? I think this causes the issue.

```
const withContentCollections = createContentCollectionPlugin({
  configPath: './content-collections/index.ts',
})
nextConfig = withContentCollections(nextConfig)
``

**Comment by sdorra** [comment]
Ok, I can reproduce it. I don't understand it, but I can reproduce it.

**Comment by sdorra** [comment]
@wottpal should be fixed with version 0.2.4 of `@content-collections/next`.

**Comment by wottpal** [comment]
Can confirm it's fixed now 🫡 You're a machine, even on Saturdays!
